File: app/src/main/java/fixtures/SayHelloFixture.java
package fixtures;

import acm.androidfitnesseexample.MessageInterface;
import acm.androidfitnesseexample.MessagePresenter;

/**
 * Created in February.
 */
public class SayHelloFixture implements MessageInterface {

    private String mMessage;
    private final MessagePresenter mMessagePresenter;

    public SayHelloFixture() {
        mMessagePresenter = new MessagePresenter(this);
    }

    public void iSay(String text) {
        mMessagePresenter.read(text);
    }

    public void waitingForAnswer() {
        mMessagePresenter.produceAnswer();
    }

    public String answer() {
        return mMessage;
    }

    @Override
    public void showMessage(String text) {
        mMessage = text;
    }
}

File: app/src/main/java/fixtures/SayHelloFixture.kt
package fixtures

import acm.androidfitnesseexample.MessageInterface
import acm.androidfitnesseexample.MessagePresenter

class SayHelloFixture : MessageInterface {

    private var mMessage: String? = null

    private val mMessagePresenter: MessagePresenter = MessagePresenter(this)

    fun iSay(text: String?) {
        mMessagePresenter.read(text)
    }

    fun waitingForAnswer() {
        mMessagePresenter.produceAnswer()
    }

    fun answer(): String? {
        return mMessage
    }

    override fun showMessage(text: String) {
        mMessage = text
    }

}